### DISCUSSION TITLE
System loses calibration quickly when limiting Low SOC via ESS

### DISCUSSION THREAD

**Discussion by HotteX** [Louisvdw/dbus-serialbattery#435] [Q&A] [answered]
Hi,

I limited my discharge SOC to first 10%, then 5% SOC via ESS. Thought being: Now in Winter I cannot fully charge the battery anyways, so I keep it at a higher SOC, unless a powerloss occurs in wich case the system has that 10% buffer. 

I am not even sure if the Venus OS or Serial-Battery is the cause for that serious and quick decalibration, but I would love to be enlightened. 
I run Venus 3.00-7 and Serial Battery 0.14.3 with a JBD-BMS.

This setup works fine for a couple of days until I noticed (first with 10%, then the same with 5% minimum SOC), that the batteries have more and more real SOC, when the system thinks it's only 5%. This only gets reset when I allow discharge to 0%. Then the % falls to 0% and THEN the battery gets discharged until low voltage is reached. Only problem being, that discharge current is limited to 5A because of the virtual 0% SOC. 

If I keep the  5% SOC limit, the voltage of the battery climbs slowly upwards when the Venus OS cuts of at 5% SOC left.

This does not happen when I don't limit the SOC, then the SOC-Counter behaves "accurate" (it feels predictable).

Screenshot is such a "fake" 5%
The real 5% shows cell voltages of slighty below 3.0

![Bildschirm­foto 2023-01-26 um 09 01 03](https://user-images.githubusercontent.com/67420589/214801881-2e8a4684-3776-4c5a-9805-17bf7032a883.png)

**Comment by Louisvdw** [comment, marked as answer]
I'll have to make a video to explain all that 😄 
In short there are 3 things influencing it that create this scenario.

1. The BMS cannot detect very small currents, so while there might be a charge going in to the battery if it is to small it cannot be counted by the shunt that the battery use. This is knon as battery drift, or SOC drift.
2. Most BMS has some cell voltage range mapped to the SOC to correct this. Because of the flat voltage curve for litiums those ranges are mostly close to the empty of full voltages. You're nicely in the middle so this correction is never triggered. (for the JBD you will see those cell voltages in the settings in the app)
3. Your cell voltages is also too low to activate the balancers on your cells. So the cell drift can be more. 

All this is the reason that you see the ESS  Mode: Optimised with BatteryLife. This mode always makes sure your battery is fully charge sometime in a week so that the ballancers get activated. 
https://www.victronenergy.com/media/pg/Energy_Storage_System/en/configuration.html#UUID-9da5aea1-7c3d-d801-4a02-ae9feecf6d0f

**Reply by HotteX** [reply]
Thanks for the help. No video neccessary, I think.

1. I know that, and I guess that this issue leads to my quick drift (visible within about 5-10 days I think). With loads quickly switching between charge and discharge when the power from solar is close to the powerdrain from the house. I just thought thats what 2 tries to correct.

2. My BMS settings are 
  VoltageCap80        3300
  VoltageCap60        3250
  VoltageCap40        3200
  VoltageCap20        3100
I don't know if the BMS uses the value for CellUnderVoltage (2700) or ChargeEnd (2800) as "0"

So for the BMS the voltage should be between 20% and 40%, not 5%. But it seems not to correct that. 

3. yeah. my balancer starts higher but the celldrift is not very high (only that 15 needs a bit of help) and even the low cell is higher than 5% of its capacity. 

I have the BatteryLife deactivated. Our grid prices are pretty high, so I would rather not charge the pack just to have a reliable SOC. For me it would be absolutely ok to have the estimate that the values of 2 should give.

New things I found out while taking a closer look today: 
In the morning my System was at the state of the screenshot above (lowest Cell: 3,158V, Sum 57,51, 5%)
I lowered the Minimum SOC to 0%, the battery discharged to about 3% (I haven't written down the voltages but they where higher than real 3%) but then there was enough power from solar so it charged again. Not much, maybe 10% at the maximum because the weather was very cloudy today, but it did never reach 0%. 
Right now it is at 2% and the values look a lot more plausibe like a real 2% (Lowest Cell 2,72V, others at 2,9V, Sum 52,5V.

So whatever happened through the day, the system seems to have recalibrated its SOC shown, but it does not seem to do that when a minimum SOC is set in ESS. At least thats what it looks like.
